CMakeLists.txt

```diff
@@ -16,11 +22,66 @@ if(ALPAKA_ACC_GPU_CUDA_ENABLE)
     set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};--expt-extended-lambda")
 endif()
 
-# Use doxygen for api doc when available.
-find_package(Doxygen QUIET)
-if(DOXYGEN_FOUND)
+
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME}
+  INTERFACE
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+target_link_libraries(${PROJECT_NAME}
+  INTERFACE
+  alpaka::alpaka)
+add_library(vikunja::vikunja ALIAS ${PROJECT_NAME})
+
+
+if(${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME})
+  #########################################
+  ######### Install Configuration #########
+  #########################################
+
+  include(CMakePackageConfigHelpers)
+  write_basic_package_version_file(
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    VERSION 0.1
+    COMPATIBILITY AnyNewerVersion
+    )
+
+  include(GNUInstallDirs)
+  install(TARGETS ${PROJECT_NAME} alpaka
+    EXPORT "${PROJECT_NAME}Targets"
+    )
+
+  include(CMakePackageConfigHelpers)
+  configure_package_config_file(
+    "${PROJECT_SOURCE_DIR}/cmake/${PROJECT_NAME}Config.cmake.in"
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
+    )
+
+  install(EXPORT "${PROJECT_NAME}Targets"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
+    NAMESPACE ${PROJECT_NAME}::)
+  install(FILES
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+  install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION include)
+
+  #########################################
+  ############# Documentation #############
+  #########################################
+
+  # Use doxygen for api doc when available.
+  find_package(Doxygen QUIET)
+  if(DOXYGEN_FOUND)
     doxygen_add_docs(doc include/)
+  endif()
+
+  #########################################
+  ################ Testing ################
+  #########################################
+
+  enable_testing()
```

Comment:
Shouldn't this be configured by the user?

Reply:
I think this just enables the testing capabilities of CMake and not the tests per se. CMake should then have an option `BUILD_TESTING` which allows to enable/disable building tests.

Reply:
I added the option `BUILD_TESTING`